packages/commerce-sdk-react/src/hooks/useAuthorizationHeader.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import useAuth from './useAuth'
+
+// This feels like a bad name
+export const useAuthorizationHeader = <Arg extends {headers?: Record<string, string>}, Ret>(
```

Comment:
Basically what `useAuthenticatedClient` does, but sets the authorization header on the method headers, rather than the config headers.